moabb/datasets/base.py

```diff
@@ -243,27 +215,16 @@ def get_data(
         cache_config: dict | CacheConfig
             Configuration for caching of datasets. See ``CacheConfig``
             for details.
-        raw_pipeline: sklearn.pipeline.Pipeline | sklearn.base.TransformerMixin
-            | None
-            Pipeline that necessarily takes a mne.io.Raw as input,
-            and necessarily returns a :class:`mne.io.Raw` as output.
-        epochs_pipeline: sklearn.pipeline.Pipeline |
-            sklearn.base.TransformerMixin | None
-            Pipeline that necessarily takes a mne.io.Raw as input,
-            and necessarily returns a :class:`mne.Epochs` as output.
-        array_pipeline: sklearn.pipeline.Pipeline |
-            sklearn.base.TransformerMixin | None
-            Pipeline either takes as input a :class:`mne.Epochs` if
-            epochs_pipeline is not ``None``, or a :class:`mne.io.Raw`
-            otherwise. It necessarily returns a :func:`numpy.ndarray`
-            as output.
-            If array_pipeline is not None, each run will be a
-            dict with keys "X" and "y" corresponding respectively to the array
-             itself and the corresponding labels.
-        events_pipeline: sklearn.pipeline.Pipeline |
-            sklearn.base.TransformerMixin | None
-            Pipeline used to generate the events. Only used if
-            ``array_pipeline`` is not ``None``.
+        process_pipeline: Pipeline | None
+            Optional processing pipeline to apply to the data.
+            To generate an adequate pipeline, we recommend using
+            :func:`moabb.paradigms.base.BaseProcessing.make_process_pipelines`.
+            This pipeline will receive :class:`mne.io.BaseRaw` objects.
+            The steps names of this pipeline should be elements of :class:`StepType`.
+            According to their name, the steps should either return a
+            :class:`mne.io.BaseRaw`, a :class:`mne.Epochs`, or a :func:`numpy.ndarray`.
+            This pipeline must be "fixed" because it will not be trained,
```

Comment:
if it is not fixed, we cannot cache its results

Reply:
Not sure to agree. For example, if your preprocessing pipelines is a scaler (like a `sklearn.preprocessing.StandardScaler`), you need a fit and it doesn't affect the digest. I verified that a pipeline fitted or not keeps the same digest. A we could cache the results, as they always be the same.

I think using only transform in `moabb.datasets.base.apply_step` may be too restrictive, we could use a clone().fit_transform() without loss of generality. Here, clone avoid contamination if partial_fit is not correctly implemented, for example to avoid mixing average/std between subjects.

Reply:
We had a discussion about this: we need to be sure that there is no information leakage. Pre/post-processing could access information from several sessions or subjects, so we must ensure that no information is leaked, hence the fixed pipelines.